packages/react/src/components/Forms/Checkbox/index.tsx

```diff
@@ -1,17 +1,102 @@
-import React, { HTMLProps, useEffect, useRef } from "react";
+import React, {
+  InputHTMLAttributes,
+  PropsWithChildren,
+  useEffect,
+  useState,
+} from "react";
+import classNames from "classnames";
+import { Field, FieldProps } from "components/Forms/Field";
+
+type Props = InputHTMLAttributes<HTMLInputElement> &
+  FieldProps & {
+    indeterminate?: boolean;
+    label?: string;
+  };
 
 export const Checkbox = ({
   indeterminate,
   className = "",
-  ...rest
-}: { indeterminate?: boolean } & HTMLProps<HTMLInputElement>) => {
-  const ref = useRef<HTMLInputElement>(null!);
+  checked,
+  label,
+  text,
+  rightText,
+  state,
+  ...props
+}: Props) => {
+  const [value, setValue] = useState<boolean>(!!checked);
 
   useEffect(() => {
-    if (typeof indeterminate === "boolean") {
-      ref.current.indeterminate = !rest.checked && indeterminate;
-    }
-  }, [ref, indeterminate]);
+    setValue(!!checked);
+  }, [checked]);
+
+  return (
+    <label
+      className={classNames("c__checkbox", {
+        "c__checkbox--disabled": props.disabled,
+      })}
+    >
+      <Field text={text} rightText={rightText} compact={true} state={state}>
+        <div className="c__checkbox__container">
+          <div className="c__checkbox__wrapper">
+            <input
+              type="checkbox"
+              className={className}
+              onChange={(e) => setValue(e.target.checked)}
+              {...props}
+              checked={value}
+            />
+            {indeterminate ? <Indeterminate /> : value && <Checkmark />}
+          </div>
+          {label && <div className="c__checkbox__label">{label}</div>}
```

Comment:
We should use a label element here. Furthermore, for a11y purpose I believe label should before the input the dom. Then we could reverse those elements in css to render them properly.